framework/auth/cas.py

```diff
@@ -149,6 +149,20 @@ def _parse_profile(self, raw):
             resp.attributes[attribute] = data['attributes'][attribute]
         return resp
 
+    def revoke_application_tokens(self, client_id, client_secret):
+        """Revoke all tokens associated with a given CAS client"""
+        url = furl.furl(self.BASE_URL)
+        url.path.segments.extend(('oauth2', 'revoke'))
+        url.args['client_id'] = client_id
+        url.args['client_secret'] = client_secret
```

Comment:
Even though it's server-to-server, I'm going to say no to my own code on that one and change it. (though there is an ancillary question of whether CAS should respond to secrets in query parameters at all)